File: persistence/src/main/java/com/pij/zworkout/persistence/xml/PersistenceModule.java
package com.pij.zworkout.persistence.xml;

import com.pij.zworkout.persistence.api.WorkoutSerializerService;

import dagger.Module;
import dagger.Provides;

/**
 * <p>Created on 04/03/2018.</p>
 *
 * @author Pierrejean
 */
@Module
public class PersistenceModule {

//    @Provides
//    WorkoutSerializerService provideDummyWorkoutSerializerService() {
//        return new DummyWorkoutSerializerService();
//    }

    @Provides
    WorkoutSerializerService provideXmlWorkoutSerializerService() {
        return new XmlWorkoutSerializerService();
    }
}

File: persistence/src/main/java/com/pij/zworkout/persistence/xml/PersistenceModule.kt
package com.pij.zworkout.persistence.xml

import com.pij.zworkout.persistence.api.WorkoutSerializerService

import dagger.Module
import dagger.Provides
import org.simpleframework.xml.core.Persister

/**
 *
 * Created on 04/03/2018.
 *
 * @author Pierrejean
 */
@Module
class PersistenceModule {

    //    @Provides
    //    WorkoutSerializerService provideDummyWorkoutSerializerService() {
    //        return new DummyWorkoutSerializerService();
    //    }

    @Provides
    internal fun provideXmlWorkoutSerializerService(): WorkoutSerializerService {
        return XmlWorkoutSerializerService(Persister())
    }
}
